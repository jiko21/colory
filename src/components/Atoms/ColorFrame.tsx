import { Flex, View } from '@adobe/react-spectrum';
import React from 'react';

interface Props {
  color: string;
  width?: string;
  height?: string;
}

const ColorFrame: React.FC<Props> = React.memo(({ children, color, width, height }) => (
  <Flex
    UNSAFE_style={{
      backgroundColor: color,
    }}
    width={width}
    height={height}
    alignItems="center"
    justifyContent="center"
  >
    <View>{children}</View>
  </Flex>
));

export default ColorFrame;
