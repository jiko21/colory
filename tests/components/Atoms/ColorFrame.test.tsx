import ColorFrame from '@/components/Atoms/ColorFrame';
import { render } from '@testing-library/react';
import React from 'react';

describe('components/Atoms/ColorFrame.tsx', () => {
  it('correctly render', () => {
    const component = render(<ColorFrame color={'#ff0000'}>children</ColorFrame>);
    expect(component.container).toMatchSnapshot();
  });
});
