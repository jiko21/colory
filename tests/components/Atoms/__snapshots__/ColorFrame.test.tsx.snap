// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Atoms/ColorFrame.tsx correctly render 1`] = `
<div>
  <div
    class="_flex_ffbb7"
    style="background-color: rgb(255, 0, 0); align-items: center; justify-content: center;"
  >
    <div>
      children
    </div>
  </div>
</div>
`;
